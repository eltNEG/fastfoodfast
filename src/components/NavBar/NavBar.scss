@import '../../styles/styles';

.nav-bar.ui.menu {
  background-color: $dark-blue;
  border-radius: 0;
  color: $white;
  padding: 0 5%;
  margin: 0;
  a {
    color: $white;
  }
  .mobile.right.menu {
    display: none;
  }
  h1{
      font-size: 1.8rem;
  }
}

.nav-bar + :not(.main) + .footer{
    margin-top: 80vh;
}

@include mobile {
  .nav-bar.ui.menu {
    .desktop.right.menu {
      display: none;
    }

    h1 {
      font-size: 1.3rem;
    }
    .mobile.right.menu {
      display: initial;
      .accordion.ui {
        position: relative;
      }
      a {
        color: $dark-blue;
      }
    }
    .accordion-content {
      position: absolute;
      padding-bottom: 0;
      right: -4rem;
      width: 100vw;
      top: -1rem;
      background-color: $dirty-white;
      text-align: center;
      z-index: 4;
    }

    i.big.icon,
    i.big.icons {
      font-size: 1.4rem;
      margin-top: 0.5rem;
    }
  }
}
