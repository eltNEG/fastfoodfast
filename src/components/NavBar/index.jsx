import React from 'react';
import { Menu, Accordion, Icon } from 'semantic-ui-react';
import { Link } from 'react-router-dom';
import './NavBar.scss';
import { HOME_PATH, LOGIN_PATH, REGISTER_PATH } from '../../constants';

class NavBar extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      active: false
    };
  }

<<<<<<< HEAD
  toggleMenu = () => {
    const { active } = this.state;
    this.setState({ active: !active });
  };

  renderMenuItemLink = (path, text) => (
    <Menu.Item>
      <Link to={path}>
        <div>{text}</div>
      </Link>
    </Menu.Item>
  );

  renderDesktopMenu = () => (
    <Menu.Menu className="desktop" position="right">
      {this.renderMenuItemLink(REGISTER_PATH, 'Register')}
      {this.renderMenuItemLink(LOGIN_PATH, 'Login')}
    </Menu.Menu>
  );

=======
  renderDesktopMenu = () => (
    <Menu.Menu className="desktop" position="right">
      <Menu.Item>
        <Link to={REGISTER_PATH}>
          <div>Register</div>
        </Link>
      </Menu.Item>
      <Menu.Item>
        <Link to={LOGIN_PATH}>
          <div>login</div>
        </Link>
      </Menu.Item>
    </Menu.Menu>
  );

  toggleMenu = () => {
    const { active } = this.state;
    this.setState({ active: !active });
  };

>>>>>>> [ft-#164048770] Implement header and footer
  renderMobileMenu = () => {
    const { active } = this.state;
    return (
      <Menu.Menu position="right" className="mobile">
        <Accordion as="item">
          <Accordion.Title active={active} onClick={this.toggleMenu}>
            <Icon size="big" inverted name="bars" />
          </Accordion.Title>
          <Accordion.Content as="menu" className="accordion-content" active={active}>
<<<<<<< HEAD
            {this.renderMenuItemLink(REGISTER_PATH, 'Register')}
            {this.renderMenuItemLink(LOGIN_PATH, 'Login')}
=======
            <Menu.Item>
              <Link to={REGISTER_PATH}>
                <div>Register</div>
              </Link>
            </Menu.Item>
            <Menu.Item>
              <Link to={LOGIN_PATH}>
                <div>login</div>
              </Link>
            </Menu.Item>
>>>>>>> [ft-#164048770] Implement header and footer
          </Accordion.Content>
        </Accordion>
      </Menu.Menu>
    );
  };

  render() {
    return (
<<<<<<< HEAD
      <Menu borderless className="nav-bar">
=======
      <Menu fixed borderless className="nav-bar">
>>>>>>> [ft-#164048770] Implement header and footer
        <Menu.Menu position="left">
          <Menu.Item>
            <Link to={HOME_PATH}>
              <h1>FastFoodFast</h1>
            </Link>
          </Menu.Item>
        </Menu.Menu>
        {this.renderDesktopMenu()}
        {this.renderMobileMenu()}
      </Menu>
    );
  }
}

export default NavBar;
