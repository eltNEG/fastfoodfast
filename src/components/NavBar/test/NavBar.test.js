import React from 'react';
import { shallow } from 'enzyme';
import NavBar from '..';

describe('NavBar.jsx', () => {
  describe('NavBar snapshot', () => {
    it('should match NavBar.jsx snapshot', () => {
      const wrapper = shallow(<NavBar />);
      expect(wrapper).toMatchSnapshot();
    });
  });
  describe('toggleMenu', () => {
      it('should set state', () => {
        const wrapper = shallow(<NavBar />);
        wrapper.instance().toggleMenu()
        expect(wrapper.state().active).toEqual(true);
      })
  })
});
