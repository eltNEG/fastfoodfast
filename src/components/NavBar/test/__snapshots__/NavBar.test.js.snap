// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar.jsx NavBar snapshot should match NavBar.jsx snapshot 1`] = `
<Menu
  borderless={true}
  className="nav-bar"
>
  <MenuMenu
    position="left"
  >
    <MenuItem>
      <Link
        replace={false}
        to="/"
      >
        <h1>
          FastFoodFast
        </h1>
      </Link>
    </MenuItem>
  </MenuMenu>
  <MenuMenu
    className="desktop"
    position="right"
  >
    <MenuItem>
      <Link
        replace={false}
        to="/register"
      >
        <div>
          Register
        </div>
      </Link>
    </MenuItem>
    <MenuItem>
      <Link
        replace={false}
        to="/login"
      >
        <div>
          Login
        </div>
      </Link>
    </MenuItem>
  </MenuMenu>
  <MenuMenu
    className="mobile"
    position="right"
  >
    <Accordion
      as="item"
    >
      <AccordionTitle
        active={false}
        onClick={[Function]}
      >
        <Icon
          as="i"
          inverted={true}
          name="bars"
          size="big"
        />
      </AccordionTitle>
      <AccordionContent
        active={false}
        as="menu"
        className="accordion-content"
      >
        <MenuItem>
          <Link
            replace={false}
            to="/register"
          >
            <div>
              Register
            </div>
          </Link>
        </MenuItem>
        <MenuItem>
          <Link
            replace={false}
            to="/login"
          >
            <div>
              Login
            </div>
          </Link>
        </MenuItem>
      </AccordionContent>
    </Accordion>
  </MenuMenu>
</Menu>
`;
